webapp/app/js/filters.js

```diff
@@ -39,11 +39,25 @@ treeherder.filter('stripHtml', function() {
 
 treeherder.filter('linkifyBugs', function() {
     return function(input) {
-        var re = new RegExp('(?:Bug (\\d+))', 'ig');
-        var str = input || '';
-        return str.replace(re,
-            '<a href="https://bugzilla.mozilla.org/show_bug.cgi?id=$1" target="_blank" title="Bug $1">$1</a>'
-        );
+        if (input) {
+            var re = /.*-- ([0-9]+)|.*bug-([0-9]+)|.*Bug ([0-9]+)/ig; //Matches 3 paterrns "<-- XXX" , "bug-XXX" and "Bug XXX"
+            var bug_matches = re.exec(input);
+            var pr_matches = /PR#([0-9]+)/i.exec(input);
+            if (pr_matches && bug_matches ) { //Check if there are any matches, if not return the 'input'
+                var bug_match = bug_matches[1] || bug_matches[2] || bug_matches[3];
+                var bug_url = "https://bugzilla.mozilla.org/show_bug.cgi?id="+bug_match;
+                var pr_url = "https://github.com/mozilla-b2g/gaia/pull/" + pr_matches[1];
+                var str = input;
+                var pr_hyperlink ='<a href="'+pr_url+'">'+pr_matches[1]+'</a>';
+                str = str.replace(pr_matches[1],pr_hyperlink);
+                var bug_hyperlink ='<a href="'+bug_url+'">'+bug_match+'</a>';
+                str = str.replace(bug_match,bug_hyperlink);
```

Comment:
Did you test this to be sure it works? Also, you should have a space between ',' and the next argument when calling a function. You should also space expressions like 'foo'+'bar' to 'foo' + 'bar'.